pkg/controllers/v1alpha1/fluidapp/implement.go

```diff
@@ -0,0 +1,122 @@
+/*
+  Copyright 2022 The Fluid Authors.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+*/
+
+package fluidapp
+
+import (
+	"github.com/fluid-cloudnative/fluid/pkg/common"
+	"github.com/fluid-cloudnative/fluid/pkg/utils/kubeclient"
+	"github.com/go-logr/logr"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/client-go/tools/record"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type FluidAppReconcilerImplement struct {
+	client.Client
+	Log      logr.Logger
+	Recorder record.EventRecorder
+}
+
+func NewFluidAppReconcilerImplement(client client.Client, log logr.Logger, recorder record.EventRecorder) *FluidAppReconcilerImplement {
+	r := &FluidAppReconcilerImplement{
+		Client:   client,
+		Log:      log,
+		Recorder: recorder,
+	}
+	return r
+}
+
+func (i *FluidAppReconcilerImplement) shouldReconcile(ctx reconcileRequestContext) bool {
+	pod := ctx.pod
+	if pod == nil {
+		return false
+	}
+
+	// ignore if restartPolicy is Always
+	if pod.Spec.RestartPolicy == corev1.RestartPolicyAlways {
+		return false
+	}
+
+	// ignore if no fuse container
```

Comment:
I have two suggestions:

1. I think we should check the labels `fuse`
2. Can we put check logic in the eventHandler   https://github.com/fluid-cloudnative/fluid/blob/master/pkg/ctrl/watch/runtime.go#L30. It can reduce the queue length.